src/ConnectionPool.jl

```diff
@@ -459,14 +459,14 @@ function keepalive!(tcp)
     return
 end
 
-function getconnection(::Type{TCPSocket},
+function getconnection(::Type{Sockets.TCPSocket},
```

Comment:
Can we just import Sockets into the module rather than spelling out out everywhere?

Reply:
Either way; I kind of like spelling it out just for future reference of where this thing is actually defined; "oh, in the Sockets stdlib package", as opposed to assuming it's in Base, or having to hunt through imports.

Reply:
I'd agree if it was `Sockets.TCP`. But as it is, I don't think this is an improvement, and a small diff is a good diff.

Reply:
I like `Sockets.TCP`; implemented

Reply:
> agree if it was Sockets.TCP

... but it isn't.

I'm not sure that evaling a type abbreviation into a stdlib module will be seen as a good thing.
Now if people go looking for `TCP` in the `Sockets` package they won't find it, and they'll have to look in HTTP.jl/src/compat.jl to find it.
Are you intending a PR against stdlib/Sockets to change it there too?

To me, the fact that `TCPSocket` is exported by the stdlib/Sockets package indicates that `TCPSocket` is intended to be a symbol in the global namespace for modules that are using Sockets. I really think we should just import the symbol and leave the code that uses it as-is.

Things like `HTTP.Message` make sense because they are not exported, and you can imagine having an `SMTP.Message` and an `SMS.Message`. But I just don't see that `Sockets.TCP` fits that pattern. I would expect a type called `TCP` (even in the contact of a `Sockets` module) to be a type that represents the TransferControlProtocol, not an instance of a Socket that implements that protocol. e.g. `TCP <: AbstractProtocol`, `TCPSocket <: AbstractSocket`. `Internet.TCP.Socket <: AbstractSocket` would make sense, but `Sockets.TCP` really doesn't.

Stuff like `Sockets.getsockname` also just seems more verbose for not gain.

One day, if someone refactors the `Sockets` module to use terser names that are intended to be prefixed by the module name, then lets apply that pattern, but for now, in a PR titled "0.7 compat", It really seems best to put `using Sockets` in `compat.jl` and `using ..Sockets` in modules that need it.

Reply:
I don't have the time or care to keep worrying about this; to me, it's clearer to preface everything w/ `Sockets` because it would be easy for a naive user to assume that `HTTP` itself defines a `getsockname` or `IPAddr` type address. At the end of the day, it's what, another 8 characters or so? Just not worth worrying about IMO. Feel free to change as you'd like, I really don't mind with whatever you'd like to go w/, but I'm going to merge this for now and you can change to whatever you'd like since you seem to have stronger feelings on the matter.